### ISSUE TITLE
"Clear all except" option

### ISSUE THREAD

**Issue by maverick74** [mrdokenny/History-AutoDelete#37]
Would be good to have an option that, instead of blacklisting domains/urls users could whitelist the ones that can show up in history, deleting all others

**Comment by mvevitsis** [comment]
You can sort of make it work this way. Set "keep history for" to zero, and it will delete all history every hour and upon opening chrome.

**Comment by maverick74** [comment]
But, there are some history i would like to keep... and this way it will delete all...

**Comment by RaitaroH** [comment]
I was looking for some kind of whitelist addon myself. This should be a feature. I mean you have this behavior in cookies auto delete extensions. I don't see why it should't be allowed in this one.

**Comment by ke-d** [comment]
Eventually I'll start work on HAD again when I have time.

**Comment by yapja** [comment]
Was this ever implemented?

**Comment by maverick74** [comment]
@Yapja nop!
The only add-on I'm aware of that allows you to do this is: 

History Zebra 
( https://addons.mozilla.org/en-US/firefox/addon/history-zebra/ )

It works pretty well, but the GUI is not great.

**Comment by yapja** [comment]
@ maverick74 That looks promising. I'll disable History AutoDelete until it's implemented.

BTW, do you know if History Zebra can do something like "delete all history from reddit.com except for reddit.com/r/insertsubreddithere"?

**Comment by maverick74** [comment]
@yapja i don't believe so, because it can only work with white list or with black list.

It can delete everything but reddit.com/r/insertsubreddithere but to delete just reddit.com except for reddit.com/r/insertsubreddithere it would require it to include reddit.com on the Black List and reddit.com/r/insertsubreddithere on the white list (and the whitelist to overpower blacklist).

Meanwhile, last night i made a few searches about the topic and found: History Delete ( https://addons.mozilla.org/en-US/firefox/addon/delete-history/ ) that might solve all the cases
